src/python/pants/build_graph/build_configuration.py

```diff
@@ -166,6 +167,10 @@ def register_rules(self, rules):
                               if rule.dependency_optionables}
     self.register_optionables(dependency_optionables)
 
+  def register_target_adaptors(self, adaptors):
+    # TODO: type annotations and runtime validation?
```

Comment:
Did you mean to address this TODO before merging? (the type annotations at least sounds like something that can be done without much effort)

Reply:
Yes!